### PR TITLE
docs: Initial setup of documentation website.

### PR DESCRIPTION
**Pull request by butler54**
Signed-off-by: Chris Butler <chris@thebutlers.me>

## Types of changes

- \[ \] Bug fix (non-breaking change which fixes an issue)
- \[ \] New feature (non-breaking change which adds functionality)
- \[ \] Breaking change (fix or feature that would cause existing functionality to change)
- \[x\] My code follows the code style of this project.
- \[x\] My change requires a change to the documentation.
- \[x\] I have updated the documentation accordingly.
- \[ \] I have added tests to cover my changes.
- \[x\] All new and existing tests passed.
- \[x\] All commits are signed.

This is an initial setup for a documentation website as described in #83. Given the complexity of our current build process I've chosen so far the simplest possible path - that is a github 'project page' (as opposed to an organizational page OR read the docs).

Note that some of the api docs will not be resolved correctly until the models are updated in #233 to produce the cleanest possible outcomes.

NOTE: Draft at this stage and expecting substantial feedback will be required to get this where we want it to be.

### CODE REVIEW

**Comment by butler54** [comment]
Ready for review.

**Review by aNebula — CHANGES_REQUESTED** [review]
We need a small change in the doc for `trestle merge` to take only -e instead of -f and -d. I will make changes in the spec through the #295 .

File: docs/cli.md [hunk 1]
```diff
@@ -256,27 +256,12 @@ In the near future, `trestle split` should be smart enough to figure out which j
...
-  In order to determine which elements the user can merge at the level the command is being executed, the following command can be used:
-  `trestle merge -l` which would be the same as `trestle merge --list-available-elements`
+For example, in the command `trestle merge -e catalog.metadata`, executed in the same directory where `catalog.json` or splitted `catalog` directory exists, the property `metadata` from `metadata.json` or `metadata.yaml` would be moved/merged into `catalog.json`. If the `metadata` model has already been split into smaller sub-component models previously, those smaller sub-components are first recusively merged into `metadata`, before merging `metadata` subcomponent into `catalog.json`. To specify merging every sub-component split from a component, `.*` can be used. For example, `trestle merge -e catalog.*` command, issued from the directory where `catalog.json` or`catalog` directory exists, will merge every single sub-component of that catalog back into the `catalog.json`.
```
Comment (aNebula) [inline docs/cli.md]: @butler54 Please review these and let me know if they make sense

File: docs/specifications/trestle-spec.md [hunk 2]
```diff
@@ -495,27 +495,12 @@ An example of a sequence of trestle split and merge commands and the correspondi
...
-  In order to determine which elements the user can merge at the level the command is being executed, the following command can be used:
-  `trestle merge -l` which would be the same as `trestle merge --list-available-elements`
+For example, in the command `trestle merge -e catalog.metadata`, executed in the same directory where `catalog.json` or splitted `catalog` directory exists, the property `metadata` from `metadata.json` or `metadata.yaml` would be moved/merged into `catalog.json`. If the `metadata` model has already been split into smaller sub-component models previously, those smaller sub-components are first recusively merged into `metadata`, before merging `metadata` subcomponent into `catalog.json`. To specify merging every sub-component split from a component, `.*` can be used. For example, `trestle merge -e catalog.*` command, issued from the directory where `catalog.json` or`catalog` directory exists, will merge every single sub-component of that catalog back into the `catalog.json`.
```
Comment (aNebula) [inline docs/specifications/trestle-spec.md:503]: @butler54 same changes here.

**Review by aNebula — APPROVED** [review]
LGTM

**Pull request merged**